contracts/token/psp35/psp35.rs

```
@@ -282,40 +272,109 @@ impl<T: PSP1155Storage + Flush> PSP1155Internal for T {
         self.get().balances.get((&id, &owner)).unwrap_or(0)
     }
 
-    default fn _is_approved_for_all(&self, account: AccountId, operator: AccountId) -> bool {
-        self.get().operator_approval.get((&account, &operator)).unwrap_or(false)
-    }
-
     default fn _increase_receiver_balance(&mut self, to: AccountId, id: Id, amount: Balance) {
         let to_balance = self.get_mut().balances.get((&id, &to)).unwrap_or(0);
         self.get_mut().balances.insert((&id, &to), &(to_balance + amount));
     }
 
-    default fn _decrease_sender_balance(
-        &mut self,
-        from: AccountId,
-        id: Id,
-        amount: Balance,
-    ) -> Result<(), PSP1155Error> {
+    default fn _decrease_sender_balance(&mut self, from: AccountId, id: Id, amount: Balance) -> Result<(), PSP35Error> {
         let balance = self.balance_of(from, id);
+
         if balance < amount {
-            return Err(PSP1155Error::InsufficientBalance)
+            return Err(PSP35Error::InsufficientBalance)
         }
 
         self.get_mut().balances.insert((&id, &from), &(balance - amount));
         Ok(())
     }
 
+    default fn _get_allowance(&self, owner: AccountId, operator: AccountId, id: Option<Id>) -> Balance {
+        let approval_for_all = self.get().operator_approvals.get(&(owner, operator, None)).unwrap_or(0);
+        let approval_for_token = self.get().operator_approvals.get(&(owner, operator, id)).unwrap_or(0);
+
+        return if approval_for_all != 0 {
+            approval_for_all
+        } else {
+            approval_for_token
+        }
+    }
+
+    fn _approve_for(&mut self, operator: AccountId, token: Option<(Id, Balance)>) -> Result<(), PSP35Error> {
+        let caller = Self::env().caller();
+
+        if caller == operator {
```

this check is not needed imo,
if so we should return a different error